src/AutoRest.CSharp/LowLevel/Generation/LowLevelModelWriter.cs

```diff
@@ -48,9 +49,19 @@ private static void WritePublicConstructor(CodeWriter writer, ConstructorSignatu
             {
                 var initializedFields = WriteFieldsInitialization(writer, signature, model);
                 // TODO: Add IReadOnlyDictionary
-                foreach (var field in model.Fields.Where(f => !initializedFields.Contains(f) && TypeFactory.IsReadOnlyList(f.Type)))
+                foreach (var field in model.Fields.Where(f => !initializedFields.Contains(f)))
                 {
-                    writer.Line($"{field.Name:I} = new List<{field.Type.Arguments[0]}>(0).AsReadOnly();");
+                    if (TypeFactory.IsList(field.Type))
```

Comment:
I think we should use `FieldDeclaration.DefaultValue` here, and `DefaultValue` itself should be set in `ModelTypeProvider`

Reply:
Then the whole block would be something like:
```cs
foreach (var field in model.Fields.Where(f => !initializedFields.Contains(f)))
{
    if (field.DefaultValue is not null) 
    {
        writer.Line($"{field.Name:I} = {field.DefaultValue}");
    }
}
```

Reply:
I tried `DefaultValue`, and the problem is that it will generate something like below:
```csharp
public IList<string> OptionalStringList { get; } = new ChangeTrackingList<string>();
```
The relevant codes are https://github.com/archerzz/autorest.csharp/blob/34cbccd14ea91fc645dfa1b23f9e8b9e8d924c89/src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs#L113

Reply:
Update that line in CodeWriterExtensions.cs to `if ((modifiers.HasFlag(FieldModifiers.Const) || modifiers.HasFlag(FieldModifiers.Static)) && field.DefaultValue != null)`. We should not initialize instance fields this way.